### PR TITLE
gimap suggested structure and schema

### PR DESCRIPTION
**Pull request by cansavvy**
# Description

 

This PR is meant to rehash the overall schema for this code -- how do we see this code being used? 

Keep in mind this will certainly change somewhat from what I have set up here, because the details of the code have not yet been accounted for. 

But this is meant to be a starting point for discussion so we can start with an idea of how users would use gimap -- and adjust as needed when the development of the functions is actually being done. 

This schema and usage is inspired by popularly used bioinformatics R packages: 

- We'd like to take [experimental design information like DESeq2](https://www.bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html) does this and it returns test results! 
- [SingleCellExperiments](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html) store and transform data but in a way that has a specialized dataset structure -- here we are trying to do the same idea. This allows users to not have to keep track of all the different pieces of information but still allows them and us to access it when the next step needs to be performed.
-  The tidyverse allows people to construct pipelines by using `%>%` or more recently `|>` this schema also allows people to do this as well since resaving the data at every step isn't always necessary. 
- Much like how [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) works, the qc step doesn't make actual changes to the data but prints out a report that users should be highly encouraged to consult, but not required to use if they don't want to. 

## Here's the big picture idea: 

These are from https://docs.google.com/presentation/d/1J-S6UpYB2IADRekObkM3HI4qEdsWvR9en780uNwiD0c/edit#slide=id.p

gimap schema

gimap schema (1)

## What does this mean for users? 

In this idea the usage would look like this for a user: 

```
# USERS SET UP THEIR DATA (this would be variable but use setup_data()

# Highly recommended but not required
run_qc(gimap_dataset)

gimap_dataset <- gimap_dataset %>%
  gimap_filter() %>%
  gimap_annotate() %>%
  calc_lfc() %>%
  calc_gi()
```
Each of the steps would optionally have arguments at each of them, but this would be the simplest version of the pipeline. 

## Which steps are required? 
- Setup is absolutely required. 
- QC is optional but recommended.
- Filtering is *technically* optional but again recommended.
- Annotation is required IF log fold change is to be calculated. 
- Log fold change is required if Genetic interactions are to be calculated

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -94,3 +102,54 @@ make_out_dir <- function(out_dir) {
...
 }
+
+#' Get file path to an default credentials RDS
+#' @export
+#' @return Returns the file path to folder where the example data is stored
+example_data_folder <- function() {
+  file <- list.files(
+    pattern = "example_data.md",
+    recursive = TRUE,
+    system.file("extdata", package = "metricminer"),
```
Comment (kweav) [inline R/utils.R:113]: Should package be "gimap"?

**Comment by kweav** [comment]
Big picture/schema/usage/description all look good to me -- thanks!

**Comment by howardbaek** [comment]
Looks good to me!

File: R/utils.R (hunk 1 above)
Comment (cansavvy) [inline R/utils.R:113]:
```suggestion
    system.file("extdata", package = "gimap"),
```

**Pull request merged**